.github/workflows/ci.yml

```diff
@@ -1,29 +1,120 @@
 on: [push, pull_request]
+  push:
+    branches:
+      - main
+      - test-github-actions
+  pull_request:
+    branches:
+      - main
 
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
+        os: [ubuntu, windows, macos]
         python-version: ["3.7", "3.8", "3.9", "3.10", "3.11", "3.12"]
     steps:
       - uses: actions/checkout@v3
+
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v3
         with:
           python-version: ${{ matrix.python-version }}
+
       - name: Install dependencies
+        if: matrix.os == 'ubuntu'
         run: |
           sudo apt-get update
           sudo apt-get install libhdf5-serial-dev zlib1g-dev libatlas-base-dev lcov swig3.0 libpcsclite-dev
           mkdir -p ~/bin/ && ln -s /usr/bin/swig3.0 ~/bin/swig && export PATH=~/bin/:$PATH
-          pip install -e .[ci,test,yubikey] # Install with CI deps
+          pip install -e .[ci,test,yubikey]
+
       - name: setup github user
         shell: bash
         run: |
           git config --global user.name oll-bot
           git config --global user.email developers@openlawlib.org
+
       - name: Run pre-commit and test with pytest
         run: |
           pre-commit run --all-files
           pytest taf/tests
+
+      - name: Build standalone executable with Docker (Linux)
+        if: matrix.os == 'ubuntu'
+        run: |
+          docker build -t taf-linux -f .github/dockerimages/Dockerfile.linux .
+          docker cp $(docker create --rm taf-linux):/usr/src/app/taf/dist/taf taf-linux
+
+      - name: Build standalone executable with Docker (Windows)
+        if: matrix.os == 'windows'
+        run: |
+          docker build -t taf-windows -f .github/dockerimages/Dockerfile.windows .
+          docker cp $(docker create --rm taf-windows):/taf/dist/taf.exe taf-windows
+
+      - name: Build standalone executable with Docker (macOS)
+        if: matrix.os == 'macos'
+        run: |
+          docker build -t taf-macos -f .github/dockerimages/Dockerfile.macos .
+          docker cp $(docker create --rm taf-macos):/usr/src/app/taf/dist/taf taf-macos
+
+      - name: Test standalone executable (Linux)
+        if: matrix.os == 'ubuntu'
+        id: test-linux
+        run: |
+          ./taf-linux --help > actual_output.txt
+          if ! diff -q actual_output.txt expected_output.txt; then
```

Comment:
My concern here is that we would have to update expected output every time we change the cli (add, remove, rename a command group).  Just this part should be sufficient if we are checking if it was installed:
```
Usage: taf [OPTIONS] COMMAND [ARGS]...

  TAF Command Line Interface
```
We could discuss is there is more that we should test, but that does not have to be a part of this PR